andela/knights-bn-backend
# #170947690 rate accomodations

#### What does this PR do?

- Allow users to rate accommodations

#### Description of Task to be completed?

- Have the following endPoint work:

   GET     /edit/accommodation/rate/:id

#### How should this be manually tested?

- **login as any user**

       { 
	     "email": "alain.maxime@gmail.com", 
	     "password": "Niyonkuru@1" 
       }

- **view all the accommodation to get the id**

       GET    /api/v1/view/accommodations

-**Rate the accommodation**

       PATCH     /api/v1/edit/accommodation/rate/:id

                        eg: /api/v1/edit/accommodation/rate/1

- **View all to see the average rate**

       GET     http://localhost:4000/api/v1/view/accommodations

#### Any background context you want to provide?

- Only existing accommodations can be rated

#### What are the relevant pivotal tracker stories?

- [#170947690](https://www.pivotaltracker.com/story/show/170947690)

#### Screenshots (if appropriate)

-  **login as any user**

<img width="1143" alt="Screen Shot 2020-03-12 at 10 42 55" src="https://user-images.githubusercontent.com/48021034/76503272-4240ef80-644e-11ea-88dc-ca31885b570b.png">


- **view all the accommodation**

<img width="1166" alt="Screen Shot 2020-03-12 at 10 41 29" src="https://user-images.githubusercontent.com/48021034/76503291-4967fd80-644e-11ea-87ce-90acdee775bb.png">


-**Add on your rate(ADD RATE IF IT IS THE FIRST TIME RATING THAT ACCOMMODATION )**

<img width="1171" alt="Screen Shot 2020-03-11 at 17 11 18" src="https://user-images.githubusercontent.com/48021034/76503342-63a1db80-644e-11ea-88ff-b927ae9851e9.png">

-**Update your rate(IF YOU HAVE RATED THAT ACCOMMODATION BEFORE)**

<img width="1166" alt="Screen Shot 2020-03-12 at 10 41 06" src="https://user-images.githubusercontent.com/48021034/76503459-a19eff80-644e-11ea-9680-2f0a61573505.png">


-**To view your impact of your rate on the accommodation**

<img width="1166" alt="Screen Shot 2020-03-12 at 10 41 29" src="https://user-images.githubusercontent.com/48021034/76503517-bb404700-644e-11ea-8e91-9d3b5602cf1a.png">


#### Questions:

-N/A
